src/index.js

```diff
@@ -65,60 +63,63 @@ function withPlugins(fn, optsArgIdx) {
 
 const formatWithCursor = withPlugins(core.formatWithCursor);
 
-const prettier = {
-  formatWithCursor,
-
-  format(text, opts) {
-    return formatWithCursor(text, opts).formatted;
-  },
-
-  check(text, opts) {
-    const { formatted } = formatWithCursor(text, opts);
-    return formatted === text;
-  },
+function format(text, opts) {
+  return formatWithCursor(text, opts).formatted;
+}
 
-  doc,
+function check(text, opts) {
+  const { formatted } = formatWithCursor(text, opts);
+  return formatted === text;
+}
 
-  resolveConfig,
-  resolveConfigFile,
-  clearConfigCache() {
-    clearConfigCache();
-    clearPluginCache();
-  },
+function clearCache() {
+  clearConfigCache();
+  clearPluginCache();
+}
 
-  /** @type {typeof getFileInfo} */
-  getFileInfo: withPlugins(getFileInfo),
-  /** @type {typeof getSupportInfo} */
-  getSupportInfo: withPlugins(getSupportInfo, 0),
+/** @type {typeof getFileInfoWithoutPlugins} */
+const getFileInfo = withPlugins(getFileInfoWithoutPlugins);
 
-  version,
+/** @type {typeof getSupportInfoWithoutPlugins} */
+const getSupportInfo = withPlugins(getSupportInfoWithoutPlugins, 0);
 
-  util: sharedUtil,
-
-  // Internal shared
-  __internal: {
-    errors,
-    coreOptions,
-    createIgnorer,
-    optionsHiddenDefaults,
-    normalizeApiOptions,
-    normalizeCliOptions,
-    utils: {
-      arrayify,
-      getLast,
-      isNonEmptyArray,
-      partition,
-    },
+// Internal shared with cli
+const sharedWithCli = {
+  errors,
+  coreOptions,
+  createIgnorer,
+  optionsHiddenDefaults,
+  normalizeApiOptions,
+  normalizeCliOptions,
+  utils: {
+    arrayify,
+    getLast,
+    isNonEmptyArray,
+    partition,
   },
+};
 
-  /* istanbul ignore next */
-  __debug: {
-    parse: withPlugins(core.parse),
-    formatAST: withPlugins(core.formatAST),
-    formatDoc: withPlugins(core.formatDoc),
-    printToDoc: withPlugins(core.printToDoc),
-    printDocToString: withPlugins(core.printDocToString),
-  },
+const debugApis = {
+  parse: withPlugins(core.parse),
+  formatAST: withPlugins(core.formatAST),
+  formatDoc: withPlugins(core.formatDoc),
+  printToDoc: withPlugins(core.printToDoc),
+  printDocToString: withPlugins(core.printDocToString),
 };
 
-export default prettier;
```

Comment:
Does this mean that we will stop doing default exports?

Reply:
Did you mean in other files? In file only export one thing, default export should still be preferred.

I make this because I think we are exporting separate things in `index.js`, maybe other people want to bundle function from prettier too, so they can get better tree-shaking.

I don't real have a strong opinion on this change. But we should make decision carefully, because unlike other changes,  this one is public, once we decide to do this, will be hard to change in the future.

Reply:
Will be happy to hear your thoughts.

Reply:
I don't prefer default export. For tree shaking and completion in a text editor.

So I am in favor of this change. However, this is a breaking change. This PR is a user facing API change. So I think we should add the changelog for this.

Reply:
WTF... Where is the changelog... I added one, only change is `esm/standalone.mjs`,  I've update the docs. The CJS version won't fell anything. Will add again.